goodtablesio/blueprints/api.py

```diff
@@ -33,6 +33,9 @@ def record_params(setup_state):
 
 @api.app_errorhandler(Exception)
 def handle_api_errors(error):
+    # TODO: this is not really correct way to catch blueprint errors
+    # because flask doesn't support error handlers per blueprint
+    # so this error handler is global for the whole app
```

Comment:
True. We should wrap the API calls in a decorator that catches exceptions and includes this logic (or another approach that applies just to the api blueprint) rather than having a error handler.